external-api/src/repositories/estore/estore-repo.ts

```diff
@@ -0,0 +1,86 @@
+import { InsertOneResult, ObjectId, UpdateFilter, UpdateResult, WithId, WithoutId } from 'mongodb';
+import { MONGO_DB_COLLECTIONS } from '@ukef/dtfs2-common';
+import { getCollection } from '../../database';
+
+/**
+ * Finds a document in the CRON_JOB_LOGS collection based on the provided deal ID.
+ *
+ * @param {ObjectId} dealId - The unique identifier for the deal.
+ *
+ * @returns {Promise<WithId<ObjectId> | WithoutId<unknown> | null>} - A promise that resolves to the found document, or `null` if no document is found.
+ *
+ * @throws {Error} - Throws an error if the find operation fails.
```

Comment:
Based on the below, the function does not throw an error if the `find` operation fails. Instead, it catches the error, logs and returns `null`